Source/SwiftLintFramework/Rules/RuleConfigurations/LineLengthConfiguration.swift

```diff
@@ -8,34 +8,65 @@
+    static let ignoreFunctionDeclarations = LineLengthRuleOptions(rawValue: 1 << 1)
+    static let ignoreComments = LineLengthRuleOptions(rawValue: 1 << 2)
+
+    static let all: LineLengthRuleOptions = [.ignoreUrls, .ignoreFunctionDeclarations, .ignoreComments]
+}
+
 public struct LineLengthConfiguration: RuleConfiguration, Equatable {
     public var consoleDescription: String {
         return length.consoleDescription + ", ignores urls: \(ignoresURLs)"
     }
 
     var length: SeverityLevelsConfiguration
     var ignoresURLs: Bool
+    var ignoresFunctionDeclarations: Bool
+    var ignoresComments: Bool
 
     var params: [RuleParameter<Int>] {
         return length.params
     }
 
-    public init(warning: Int, error: Int?, ignoresURLs: Bool) {
+    public init(warning: Int, error: Int?, options: LineLengthRuleOptions? = []) {
         length = SeverityLevelsConfiguration(warning: warning, error: error)
-        self.ignoresURLs = ignoresURLs
+        if let options = options {
+            self.ignoresURLs = options.contains(.ignoreUrls)
+            self.ignoresFunctionDeclarations = options.contains(.ignoreFunctionDeclarations)
+            self.ignoresComments = options.contains(.ignoreComments)
+        } else {
+            self.ignoresURLs = false
+            self.ignoresFunctionDeclarations = false
+            self.ignoresComments = false
+        }
     }
 
     public mutating func apply(configuration: Any) throws {
-        if let configurationArray = [Int].array(of: configuration), !configurationArray.isEmpty {
+        if let configurationArray = [Int].array(of: configuration),
+            !configurationArray.isEmpty {
             let warning = configurationArray[0]
             let error = (configurationArray.count > 1) ? configurationArray[1] : nil
             length = SeverityLevelsConfiguration(warning: warning, error: error)
         } else if let configDict = configuration as? [String: Any], !configDict.isEmpty
-            && Set(configDict.keys).isSubset(of: ["warning", "error", "ignores_urls"]) {
+            && Set(configDict.keys).isSubset(of: ["warning",
+                                                  "error",
+                                                  "ignores_urls",
+                                                  "ignores_function_declarations",
+                                                  "ignores_comments"]) {
             let warning = configDict["warning"] as? Int ?? length.warning
             let error = configDict["error"] as? Int
             length = SeverityLevelsConfiguration(warning: warning, error: error)
             ignoresURLs = configDict["ignores_urls"] as? Bool ?? ignoresURLs
+            if let funcDec =
+                configDict["ignores_function_declarations"] as? Bool {            ignoresFunctionDeclarations = funcDec
+            }
+            ignoresComments = configDict["ignores_comments"] as? Bool ?? ignoresComments
```

Comment:
Fixed.   Also embiggened the application of the configuration dictionary so that invalid value types for a parameter will throw - added tests for these cases as well